x/liquidity/types/utils.go

```diff
@@ -88,7 +89,7 @@ func GetOfferCoinFee(offerCoin sdk.Coin, swapFeeRate sdk.Dec) sdk.Coin {
 	}
 	// apply half-ratio swap fee rate and ceiling
 	// see https://github.com/tendermint/liquidity/issues/41 for details
-	return sdk.NewCoin(offerCoin.Denom, offerCoin.Amount.ToDec().Mul(swapFeeRate.QuoInt64(2)).Ceil().TruncateInt()) // Ceil(offerCoin.Amount * (swapFeeRate/2))
+	return sdk.NewCoin(offerCoin.Denom, math.Dec(offerCoin.Amount.Mul(swapFeeRate.QuoInt64(2)).Ceil().TruncateInt())) // Ceil(offerCoin.Amount * (swapFeeRate/2)) NOTE: I COULD HAVE MADE A BUG HERE IN THE DECIMAL CONVERSION.
```

Comment:
this is the line I am unsure of, sorry 'bout that.

Definitely wanted to flag it, too, looks like it's an important one. basically, it is now looking for an cosmosdk.io/math.Int